Implementação componentes de Login e Cadastro

Yasmiinmuniz opened the pull request:
Implementação componentes de Login e Cadastro e adiciona configuração de API

- Adiciona componente Login
- Adiciona componente Cadastro
- Configura API para login e cadastro
- Atualiza App.js para gerenciar rotas e alert entre Login e Cadastro

Yasmiinmuniz committed:
Implementa componentes de Login e Cadastro com estilização consistente e adiciona configuração de API

- Adiciona componente Login
- Adiciona componente Cadastro
- Configura API para login e cadastro
- Atualiza App.js para gerenciar rotas e alert entre Login e Cadastro

Yasmiinmuniz committed:
Atualização do componente Login para usar a API e tratamento de erros

- Integração com a API: Substituído o método `onLogin` pelo `loginUser` importado de `./api`, que faz uma solicitação para a API de login.
- Armazenamento de Token: Adicionada a funcionalidade para armazenar o token JWT retornado pela API no `localStorage` após o login bem-sucedido.
- Tratamento de Erros: Implementado tratamento de erros com mensagem de erro exibida para o usuário caso o login falhe.
- Props Atualizadas: Alterado `onLogin` para `onLoginSuccess` para melhor refletir o propósito de notificar sucesso no login.

Essas mudanças melhoram a integração com a API e a experiência do usuário ao fornecer feedback apropriado sobre erros de autenticação.

luiz-davi-m commented on agiota_front_end/src/api.js:
```diff
@@ -0,0 +1,25 @@
+const API_URL = 'http://localhost:3000/users';
```
Essa rota não existe

Yasmiinmuniz committed:
Refatoração das funções de registro e login de usuários para usar Axios

- Atualizado a URL da API para `http://localhost:8080`.
- Alterado o método de comunicação com a API de `fetch` para `axios`.
- Adicionada validação de dados na função `registerUser` para verificar a presença de campos obrigatórios e campos específicos para diferentes tipos de usuários (`cliente` e `agiota`).
- Atualizada a função `registerUser` para enviar dados para o endpoint `/cadastro` e melhorado o tratamento de erros.
- Alterada a função `loginUser` para enviar a solicitação de login via `GET` com parâmetros e melhorar o tratamento de erros.

Yasmiinmuniz committed:
Alteração da linguagem de inglês para português

Yasmiinmuniz committed:
Atualização das rotas da aplicação para incluir Login e Cadastro

- Adicionadas rotas para os componentes `Login` e `Cadastro`.
- Atualizada a configuração do roteador para utilizar o `createBrowserRouter` e `RouterProvider`.
- Removida a rota para `/emprestimos` e comentada a rota para a página inicial.
- Corrigida a definição das rotas para `Login` e `Cadastro` removendo o caminho relativo incorreto.

Yasmiinmuniz committed:
Atualização do componente App para incluir navegação e ajuste de nomes

- Renomeado o componente `Cadastro` para `Register` para refletir melhor seu propósito.
- Atualizado os nomes dos parâmetros `email` e `senha` para `username` e `password` na função `handleLogin` para consistência com o novo formato da API.
- Adicionada navegação com links para as páginas de `Login` e `Cadastro`.
- Comentada a rota da página inicial e o link correspondente.

Yasmiinmuniz committed:
Refatorar página de cadastro para suportar múltiplos passos e diferentes tipos de usuário

- Adiciona navegação por etapas (step) para o formulário de cadastro.
- Inclui campos para dados pessoais, endereço e informações específicas para cada tipo de usuário (cliente ou agiota).
- Adiciona funcionalidade para buscar e preencher automaticamente o endereço com base no CEP.
- Atualiza o estado para gerenciar dados de usuário, endereço e informações específicas do papel.
- Substitui o formulário de cadastro único por um formulário dividido em etapas, melhorando a experiência do usuário.

Yasmiinmuniz committed:
Refatora o componente Login para melhorar o manejo de estados e a funcionalidade dos botões

luiz-davi-m commented on agiota_front_end/src/api.js:
```diff
@@ -0,0 +1,39 @@
+      throw new Error('Dados do agiota incompletos');
+    }
+
+    const response = await axios.post(`${API_URL}/cadastro`, user);
```
essa rota não existe. Para criar um usuário, apenas através do cliente e do agiota. Basta o olhar no insomnia as requisições que estão em: Usuario > Agiota > Criar agiota **e** Usuario > Cliente > Criar Cliente

luiz-davi-m commented on agiota_front_end/src/index.js:
```diff
@@ -2,23 +2,25 @@ import React from 'react';
         element: <App/>,
         children: [
-            { path: "/" },
-            { path: "/emprestimos", element: <Emprestimos /> }
```
voltar as rotas como estavam antes, apenas adiconar as novas que vc criou e as lógices necesárias

Yasmiinmuniz committed:
Refatorar aplicação para nova estrutura de autenticação

- Atualizado `loginUser` para `onLogin` com nova lógica de autenticação e tratamento de erros.
- Alterado `registerUser` para funções `onCreateAgiota` e `onCreateCliente` para suporte a diferentes tipos de usuários.
- Atualizado componente `App` para suportar tipos de usuários (`cliente` ou `agiota`) e adaptar a lógica de autenticação e registro.
- Modificado `Login` e `Register` para integração com nova API e gestão de tipo de usuário.
- Ajustado configuração de roteador e componentes relacionados para refletir alterações na API e na lógica de autenticação.

Yasmiinmuniz committed:
feat: atualizar estilização da tela de Login

- Aprimoramento da estilização geral da tela de Login, garantindo consistência e melhorando a experiência do usuário.
- Ajuste do estado de foco para adicionar uma sombra customizada quando o campo de entrada está ativo.
- Atualização do estilo do botão, incluindo um fundo com gradiente e efeitos de hover.
- Garantia de alinhamento e espaçamento adequados dentro do formulário para um visual limpo e moderno.

Yasmiinmuniz committed:
Refatoração do Formulário de Cadastro e Estilização

- **Separação do Componente Register:** O formulário de cadastro foi dividido em componentes individuais para melhor organização e manutenção:
  - `UsuarioForm`: Formulário para informações do usuário.
  - `EnderecoForm`: Formulário para dados de endereço.
  - `RoleForm` e `RoleSelection`: Componentes para seleção e definição de roles.

- **Estilização Modernizada:** O código foi atualizado para um estilo mais moderno, incluindo:
  - Estilização aprimorada dos componentes de formulário.
  - Reestilização do header para um visual mais atraente.

- **Atualização de Arquivos:** O `app` e o `index` foram ajustados para acomodar as novas alterações nos formulários de cadastro.

- **Separação da API:** A API foi reorganizada para melhorar a integração entre o front-end, a API e o back-end, facilitando a comunicação e manutenção.

jacksonlmp committed:
Validação de CPF

jacksonlmp reviewed (commented):
Adicionado validação no campo de CPF:
- Tamanho
- Formato
![image](https://github.com/user-attachments/assets/1b3aab6a-b36e-4c9c-bca6-8de1102622c8)

jacksonlmp committed:
Validação do campo de Senha e habilitação do botão proximo

jacksonlmp reviewed (commented):
- Validação no campo de senha (mínimo 6 digitos)
- Botão **Próximo** só habilitado se todos os campos forem preenchidos
![image](https://github.com/user-attachments/assets/2e6dfc3e-5154-4390-a6eb-0a8343fc53cd)

jacksonlmp committed:
Correção de cadastro de Agiota

jacksonlmp reviewed (commented):
Agora é possível cadastrar Agiota normalmente:
Front:
![image](https://github.com/user-attachments/assets/55abb2e1-ed7d-4f15-b9e5-f977acfb28df)
API:
![image](https://github.com/user-attachments/assets/df52c21a-9a59-4530-a6eb-c8a49a8353b7)

luiz-davi-m committed:
FIX criação de agiota e cliente

luiz-davi-m committed:
melhoran do fluxo de logout

luiz-davi-m committed:
FIX: adicionando mais informações no retorno do endpoint de login + ajustando front end para resgatar esse usuaroi

luiz-davi-m committed:
Remover front end em react
